src/Refinement/RefinedDiscreteModels.jl

```diff
@@ -0,0 +1,110 @@
+
+
+"""
+Refinement glue between two nested triangulations
+- `f2c_faces_map`     : Given a fine face gid, returns the gid of the coarse face containing it. 
+- `fcell_to_child_id` : Given a fine cell gid, returns the local child id within the coarse cell containing it.
+- `f2c_ref_cell_map`  : Ref coordinate map between the fcells and their parent ccell, i.e 
+                          for each fine cell defines Φ st. x_c = Φ(x_f)
+"""
+struct RefinementGlue{A,B,C} <: GridapType
+  f2c_faces_map::A
+  fcell_to_child_id::B
+  f2c_ref_cell_map::C
+
+  function RefinementGlue(f2c_faces_map,
+                          fcell_to_child_id,
+                          f2c_ref_cell_map)
+    A = typeof(f2c_faces_map)
+    B = typeof(fcell_to_child_id)
+    C = typeof(f2c_ref_cell_map)
+    new{A,B,C}(f2c_faces_map,fcell_to_child_id,f2c_ref_cell_map)
+  end
+end
+
+"""
+Ref coordinate map from fcell ref coords to ccell ref coords.
+Size -> Number of children per coarse cell.
+"""
+function get_f2c_ref_cell_map(reffe,ref)
+  ref_grid = Geometry.UnstructuredGrid(Visualization.compute_reference_grid(reffe,ref))
+  return get_cell_map(ref_grid)
+end
+
+"""
+Ref coordinate map between fine and coarse cells. 
+Size -> Number of fine cells. 
+"""
+function get_f2c_ref_coordinate_map(g::RefinementGlue)
+  m = Reindex(g.f2c_ref_cell_map)
+  return lazy_map(m,g.fcell_to_child_id)
+end
+
+#################################################################
+
+"""
+  RefinedDiscreteModel
+
+  `DiscreteModel` created by refining another `DiscreteModel`. 
+  The hierarchy is stored, allowing for the transfer of dofs 
+  between `FESpaces` defined on this model and its parent using a 
+  `RefinementTransferoperator`. 
+"""
+struct RefinedDiscreteModel{Dc,Dp,A<:DiscreteModel{Dc,Dp},B<:DiscreteModel{Dc,Dp},C<:RefinementGlue} <: DiscreteModel{Dc,Dp}
+  model  ::A
+  parent ::B
+  glue   ::C
+end
+
+# DiscreteModel API
+Geometry.get_grid(model::RefinedDiscreteModel)          = get_grid(model.model)
+Geometry.get_grid_topology(model::RefinedDiscreteModel) = get_grid_topology(model.model)
+Geometry.get_face_labeling(model::RefinedDiscreteModel) = get_face_labeling(model.model)
+
+# Other getters
+get_model(model::RefinedDiscreteModel)  = model.model
+get_parent(model::RefinedDiscreteModel) = model.parent
+get_glue(model::RefinedDiscreteModel)   = model.glue
```

Comment:
`get_glue` is a method already defined in Gridap. We could of course overload it (BTW, are we overloading it now "as-is"? otherwise this might be quite dangerous, and end up depending on usage context with multi-method function with a single function). However, I think the intention of this particular `get_glue` is different. While the former provides the gluing info among two consecutive models, the latter provides the gluing info among a triangulation and its background model. To-think: might it make sense to have a reserved name for this kind of glue?

Reply:
Resolved in [6698404](https://github.com/gridap/Gridap.jl/pull/838/commits/6698404320283b880190cb80ce4a879a43e2d9e1)